lib/web/apiserver.go

```diff
@@ -563,6 +565,52 @@ func (h *Handler) getUserStatus(w http.ResponseWriter, r *http.Request, _ httpro
 	return OK(), nil
 }
 
+// handleGetUserOrResetToken has two handlers:
+// - read user
+// - return reset password token
+// It has two because the expected route for reading a user overlaps with an already existing one
+// Using `GET /webapi/users/:username` invalidates the `GET /webapi/users/password/token/:token` route
+// An alternative would be using the resource's singular name `GET /webapi/user/:username` but it invalidates the `GET /webapi/user/status` route
```

Comment:
Is this a limitation in the route handling code as it feels like it should support fall through route parsing...something like:
```go
h.GET("/webapi/user/status", h.WithAuth(h.getUserStatus))
h.GET("/webapi/user/:username", h.WithAuth(h.getUser))
```

Reply:
It doesn't allow us to do that unfortunately 
https://github.com/gravitational/httprouter#named-parameters
> Note: Since this router has only explicit matches, you can not register static routes and parameters for the same path segment. For example you can not register the patterns /user/new and /user/:user for the same request method at the same time. The routing of different request methods is independent from each other.